### DISCUSSION TITLE
How to iterate section and table cells?

### DISCUSSION THREAD

**Discussion by danieltan007** [dolanmiu/docx#2163] [Q&A] [answered]
hello,
i want to iterate the section and table cell based on this data
```
interface IEkstra {
  basket?: CellValue;
  tari?: CellValue;
  voli?: CellValue;
  drumband?: CellValue;
}

interface IData {
  nis: CellValue;
  nama: CellValue;
  kelas: CellValue;
  ekstra: IEkstra & {
    pramuka: CellValue;
  };
}

const dataList: IData;
```
for iterating section is like this?
```
section : dataList.map({
return [
 {
    properties: ...
  }
]
});
```

for the table cell, i want to iterate at the the dataList.ekstra with table cell
here is my code right now (i commentted this for testing purpose): 

```
new Table({
rows: [Object.keys(item.ekstra).map((ekstra, index) => {
  return new TableRow({
    ....
  })
}),
]
});
```
i got an error: typeError: r is not iterable

is there any way to iterate the section and table cell?

**Comment by dolanmiu** [comment, marked as answer]
I don't think you're doing it right, spread it using `...`

try this:

```ts
rows: [...Object.keys(item.ekstra).map((ekstra, index) => {
  return new TableRow({
    ....
  })
}),
]
```

**Reply by danieltan007** [reply]
ok, for the iterating section is already correct way?

**Reply by dolanmiu** [reply]
it should be `sections`, but yes seems correct
